### PR TITLE
[refactor] Add destroy snode tree api

### PR DESCRIPTION
**Pull request by AmesingFlank**

### CODE REVIEW

**Review by k-ye — APPROVED** [review]
LGTM!

**Review by beginlner — APPROVED** [review]
LGTM!

File: taichi/backends/metal/metal_program.h [hunk 1]
```diff
@@ -36,6 +36,9 @@ class MetalProgramImpl : public ProgramImpl {
     metal_kernel_mgr_->synchronize();
   }
 
+  virtual void destroy_snode_tree(SNodeTree *snode_tree) override{
```
Comment (ailzhang) [inline taichi/backends/metal/metal_program.h:39]: For derived class it's okay to use `override` only ;). `virtual` is only necessary for base class. ;)

**Review by ailzhang — APPROVED** [review]
LGTM!

**Pull request merged**